custom/icds_reports/management/commands/custom_ls_data_pull.py

```diff
@@ -0,0 +1,153 @@
+        vhnd = ["pregnant_woman", "received_dpt1", "received_dpt3", "received_measles"]
+        fast_data_rows = {}
+        for location in location_details:
+            for visit in home_visit:
+                row = [
+                    location[0],
+                    location[3],
+                    location[4],
+                    "home_visit",
+                    visit,
+                    0
+                ]
+                fast_data_rows.update({f"{location[2]}_home_visit_{visit}": row})
+            for visit in vhnd:
+                row = [
+                    location[0],
+                    location[3],
+                    location[4],
+                    "vhnd_day",
+                    visit,
+                    0
+                ]
+                fast_data_rows.update({f"{location[2]}_vhnd_day_{visit}": row})
+        count = 0
+        supervisor_ids = [location[3] for location in location_details]
+        for form in forms_list:
+            supervisor_id = user_location_details[form['form']['case']['@user_id']]
+            m_type = form['form']['meeting_type']
+            if m_type == 'vhnd_day':
+                visit = form['form']['beneficiary_type_for_visit_vhnd']
+            else:
+                visit = form['form']['beneficiary_type_for_visit']
+            if supervisor_id in supervisor_ids:
+                fast_data_rows[f"{supervisor_id}_{m_type}_{visit}"][5] = fast_data_rows[f"{supervisor_id}_{m_type}_{visit}"][5] + 1
+            if count % 1000 == 0:
+                print(f"{count} forms processed ======\n")
+            count = count + 1
+        for key, value in fast_data_rows.items():
+            data_rows = data_rows + [value]
+        fout = open('/home/cchq/LS_data.csv', 'w')
+        writer = csv.writer(fout)
+        writer.writerows(data_rows)
+
+
+        print("=========starting the VHNSD data pull=====\n")
+        headers = ['state_name', 'supervisor_site_code', 'launched_status',
+                   'total_vhnsd_visit_count', 'unique_awc_visited']
+        data_rows = [headers]
+        fast_rows = {}
+        for location in location_details:
+            row = [
+                location[0],
+                location[3],
+                location[4],
+                0,
+                []
+            ]
+            fast_rows.update({location[2]: row})
+        query = FormES().domain('icds-cas').xmlns(
+            'http://openrosa.org/formdesigner/b8273b657bb097eb6ba822663b7191ff6bc276ff').submitted(
+            gte=datetime(2020, 4, 1), lt=datetime(2020, 5, 1))
+        forms_list = query.run().hits
+        count = 0
+        for form in forms_list:
+            supervisor_id = user_location_details[form['form']['case']['@user_id']]
+            awc_selected_name = form['form']['awc_selected_name']
+            fast_rows[supervisor_id][3] = fast_rows[supervisor_id][3] + 1
+            fast_rows[supervisor_id][4].append(awc_selected_name)
```

Comment:
do you want to make it a set instead of list so that unique criteria is handled by itself